client/src/components/pages/RobotInteractionCockpit/RobotInteractionCockpit.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect, useCallback, useRef } from 'react';
+import { Layout, Card, Steps, Space, Button, Typography } from 'antd';
+import HeaderNavbar from '../../content/HeaderNavbar/HeaderNavbar';
+import RobotInteractionInputSection from '../../content/RobotInteractionSections/RobotInteractionInputSection';
+import { getAllParametersForRobot } from '../../../api/variableRetrieval';
+import socket from '../../../utils/socket/socketConnections';
+import { upsert } from '../../../utils/attributeAndParamUtils';
+import styles from './RobotInteractionCockpit.module.css';
+import { isRobotExecutable } from '../../../utils/robotExecution';
+
+const { Step } = Steps;
+const { Title } = Typography;
+
+/**
+ * @description Page, where you can interact with a robot and for example enter input
+ * @category Client
+ * @component
+ */
+const RobotInteractionCockpit = (match) => {
```

Comment:
what is `match `in this context?

Reply:
This is needed for the Routing. When we click on the play button in RobotOverview a new page will open up: RobotInteractionCockpit. In RobotInteractionCockpit we need some of the information that we have in the RobotOverview page (userId & robotId). Match is an object provided by react routing that allows us to access the parameters we pass to the new page. We access these parameters in the following lines...